### PR TITLE
Base: Add a man page documenting security mitigations

### PR DESCRIPTION
**Pull request by bgianfo**
Since so much work is being put into mitigations, I thought
it would be nice to track them all in one place. This is the
start of that document.

### CODE REVIEW

File: Base/usr/share/man/man7/Mitigations.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
+## Name
+
+Mitigations - Security mitigations implemented by SerenityOS.
+
+## Description
+
+Although Serenity is a hobby operating system, the developers have put substantial effort into
+adding Security mitigations to enhance the security of the system. The goal of this document is
+to collect and describe the mitigations in one centralized place.
```
Comment (awesomekling) [inline Base/usr/share/man/man7/Mitigations.md:9]: Let's not self-deprecate about being a "hobby" operating system. Of course we are, but let's take ourselves seriously :)

**Review by awesomekling — CHANGES_REQUESTED** [review]
Great idea!

File: Base/usr/share/man/man7/Mitigations.md (hunk 1 above)
Comment (bgianfo) [reply Base/usr/share/man/man7/Mitigations.md]: Sure :+1:

File: Base/usr/share/man/man7/Mitigations.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
+## Name
+
+Mitigations - Security mitigations implemented by SerenityOS.
+
+## Description
+
+The SerenityOS developers have put substantial effort into
+integration various mitigation technologies into the system
```
Comment (awesomekling) [inline Base/usr/share/man/man7/Mitigations.md:8]: integration -> integrating

File: Base/usr/share/man/man7/Mitigations.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+Author: Andreas Kling <awesomekling@gmail.com>
+Date:   Sun Jan 5 18:00:15 2020 +0100
+
+Kernel: Start implementing x86 SMAP support
+```
+
+### Pledge
+
+[pledge](https://marc.info/?l=openbsd-tech&m=143725996614627&w=2) is a mitigation which originated from OpenBSD (originally named tame).
+It allows a program to voluntarily restrict it's access to the kernels syscall
```
Comment (awesomekling) [inline Base/usr/share/man/man7/Mitigations.md:50]: it's -> its
kernels -> kernel's

File: Base/usr/share/man/man7/Mitigations.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+surface area. The allows the program to reduce the potential attack surface area
```
Comment (awesomekling) [inline Base/usr/share/man/man7/Mitigations.md:51]: Maybe collapse the second "surface area" to just "surface"

File: Base/usr/share/man/man7/Mitigations.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+Author: Andreas Kling <awesomekling@gmail.com>
+Date:   Sat Jan 11 20:45:51 2020 +0100
+
+Kernel: Add pledge() syscall :^)
+```
+
+### Unveil
+
+[unveil](https://lwn.net/Articles/767137/) is a mitigation which originated from OpenBSD.
+It allows a program to voluntarily restrict it's access to the filesystem.
```
Comment (awesomekling) [inline Base/usr/share/man/man7/Mitigations.md:67]: it's -> its

File: Base/usr/share/man/man7/Mitigations.md [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+The allows the program to reduce the potential surface
```
Comment (awesomekling) [inline Base/usr/share/man/man7/Mitigations.md:68]: The -> It

File: Base/usr/share/man/man7/Mitigations.md [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+area available if the program in question was exploited.
+
+[It](It) was first added in the following commit, and the majority of programs were enlightened later:
```
Comment (awesomekling) [inline Base/usr/share/man/man7/Mitigations.md:71]: There seems to be a link missing here.

File: Base/usr/share/man/man7/Mitigations.md [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+Author: Andreas Kling <kling@serenityos.org>
+Date:   Sun Feb 14 17:35:07 2021 +0100
+
+Kernel: Add mechanism to make some memory read-only after init finishes
+
+```
+
+### KUBSAN (Kernel Undefined Behavior Sanitizer)
+
+Undefined behavior sanitizer dynamic analysis tool that is a feature of GCC
```
Comment (awesomekling) [inline Base/usr/share/man/man7/Mitigations.md:122]: Undefined behavior sanitizer ***is a*** dynamic analysis tool

**Review by awesomekling — CHANGES_REQUESTED** [review]

**Comment by bgianfo** [comment]
I guess I need to go back to grade school and take English again, thanks for the school'in. 😁

**Pull request merged**